src/main/scala/sparkz/core/network/peer/InMemoryPeerDatabase.scala

```diff
@@ -3,23 +3,25 @@ package sparkz.core.network.peer
 import sparkz.core.app.SparkzContext
 import sparkz.core.network.peer.PeerBucketStorage.{BucketConfig, PeerBucketStorageImpl}
 import sparkz.core.network.peer.PeerDatabase.{PeerConfidence, PeerDatabaseValue}
-
-import java.net.{InetAddress, InetSocketAddress}
-import sparkz.core.settings.NetworkSettings
+import sparkz.core.network.peer.PenaltyType.DisconnectPenalty
+import sparkz.core.persistence.StorageFilePersister.StorageFilePersisterConfig
+import sparkz.core.persistence.{MapPersister, PeerBucketPersister, PersistablePeerDatabase, StoragePersister}
+import sparkz.core.settings.SparkzSettings
 import sparkz.core.utils.{NetworkUtils, TimeProvider}
 import sparkz.util.SparkzLogging
-import sparkz.core.network.peer.PenaltyType.DisconnectPenalty
 
 import java.net.{InetAddress, InetSocketAddress}
 import java.security.SecureRandom
+import scala.collection.mutable
 import scala.concurrent.duration._
 
 /**
   * In-memory peer database implementation supporting temporal blacklisting.
   */
-final class InMemoryPeerDatabase(settings: NetworkSettings, sparkzContext: SparkzContext)
-  extends PeerDatabase with SparkzLogging {
+final class InMemoryPeerDatabase(sparkzSettings: SparkzSettings, sparkzContext: SparkzContext)
```

Comment:
I think it can keep its name since all the data structures are still in memory; I introduced a strategy that takes care of how and if persist the storages. This way the user can choose whether to use no strategy (an empty one) or the one we're using that persists to file.

Reply:
ok, got it. Looks good and clear after the last changes.